src/Psalm/Internal/Analyzer/Statements/Expression/ArrayAnalyzer.php

```diff
@@ -110,8 +109,12 @@ public static function analyze(
         }
 
         if ($array_creation_info->all_list) {
-            $array_type = new Type\Atomic\TNonEmptyList($item_value_type ?: Type::getMixed());
-            $array_type->count = count($stmt->items);
+            if (empty($array_creation_info->item_key_atomic_types)) {
+                $array_type = new Type\Atomic\TList($item_value_type ?: Type::getMixed());
+            } else {
+                $array_type = new Type\Atomic\TNonEmptyList($item_value_type ?: Type::getMixed());
+                $array_type->count = count($array_creation_info->property_types);
```

Comment:
Not sure if this is correct, but I'm also not sure what would be correct. `count($stmt-items)` is clearly wrong with unpacked arrays.
Should `count` be renamed to `min_count` as well?

Reply:
`count` is only normally set when it's known — if this any of the items are unpacked arrays then we shouldn't be setting an explicit type

Reply:
Ok, so `count` is always exact, and if exact count isn't known there's no minimum count or anything, it's just not checked at all.
It would still be good to set it when unpacking literal arrays: psalm.dev/r/fe1e86a493